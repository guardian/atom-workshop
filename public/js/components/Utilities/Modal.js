import React from 'react';
import {PropTypes} from 'prop-types';

export default class Modal extends React.Component {

    static propTypes = {
      isOpen: PropTypes.bool.isRequired,
      dismiss: PropTypes.func.isRequired,
      children: PropTypes.element.isRequired
    };

    preventClosingClick (event) {
        event.stopPropagation();
    }

    render() {
        if(!this.props.isOpen) {
            return false;
        }

        return (
            <div className="modal" onClick={this.props.dismiss}>
                <div className="modal__content" onClick={this.preventClosingClick}>
                    <div className="modal__content__header">
                        <button className="i-cross modal__dismiss" onClick={this.props.dismiss}>
                            Close
                        </button>
                    </div>
                    {this.props.children}
                </div>
            </div>
        );

    }
}
