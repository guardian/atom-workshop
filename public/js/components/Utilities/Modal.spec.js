import React from 'react';
import Modal from './Modal';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

test('Should render null when closed', () => {
  const component = renderer.create(
    <Modal isOpen={false} dismiss={() => {}}><div>Modal Contents</div></Modal>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render contents when open', () => {
  const component = renderer.create(
    <Modal isOpen={true} dismiss={() => {}}><div>Modal Contents</div></Modal>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should call dismiss function when close clicked', () => {

  const dismissFn = jest.fn();
  const modal = shallow(
    <Modal isOpen={true} dismiss={dismissFn}>
      <div>Modal Contents</div>
    </Modal>
  );
  modal.find('.modal__dismiss').simulate('click');
  expect(dismissFn).toHaveBeenCalledTimes(1);
});

test('Should call dismiss function when background clicked', () => {

  const dismissFn = jest.fn();
  const modal = shallow(
    <Modal isOpen={true} dismiss={dismissFn}>
      <div>Modal Contents</div>
    </Modal>
  );
  modal.find('.modal').simulate('click');
  expect(dismissFn).toHaveBeenCalledTimes(1);
});

test('Should not call dismiss function when content clicked', () => {

  const dismissFn = jest.fn();
  const modal = shallow(
    <Modal isOpen={true} dismiss={dismissFn}>
      <div>Modal Contents</div>
    </Modal>
  );
  modal.find('.modal__content').simulate('click', { stopPropagation() {} });
  expect(dismissFn).not.toHaveBeenCalled();
});
