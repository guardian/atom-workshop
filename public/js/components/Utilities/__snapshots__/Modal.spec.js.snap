// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render contents when open 1`] = `
<div
  className="modal"
  onClick={[Function]}
>
  <div
    className="modal__content"
    onClick={[Function]}
  >
    <div
      className="modal__content__header"
    >
      <button
        className="i-cross modal__dismiss"
        onClick={[Function]}
      >
        Close
      </button>
    </div>
    <div>
      Modal Contents
    </div>
  </div>
</div>
`;

exports[`Should render null when closed 1`] = `null`;

exports[`test Should render contents when open 1`] = `
<div
  className="modal"
  onClick={[Function]}>
  <div
    className="modal__content"
    onClick={[Function]}>
    <div
      className="modal__content__header">
      <button
        className="i-cross modal__dismiss"
        onClick={[Function]}>
        Close
      </button>
    </div>
    <div>
      Modal Contents
    </div>
  </div>
</div>
`;

exports[`test Should render null when closed 1`] = `null`;
